RecoVertex/BeamSpotProducer/plugins/BeamSpotCompatibilityChecker.cc

```diff
@@ -0,0 +1,242 @@
+    }
+
+  private:
+    double m_A;
+    double m_B;
+    double m_ErrA;
+    double m_ErrB;
+  };
+}  // namespace bscheck
+
+class BeamSpotCompatibilityChecker : public edm::global::EDAnalyzer<> {
+public:
+  explicit BeamSpotCompatibilityChecker(const edm::ParameterSet&);
+  ~BeamSpotCompatibilityChecker() override = default;
+
+  void analyze(edm::StreamID, edm::Event const&, edm::EventSetup const&) const final;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  static std::array<float, 6> compareBS(const reco::BeamSpot& BSA, const reco::BeamSpot& BSB);
+
+private:
+  // ----------member data ---------------------------
+  static constexpr int cmToum = 10000;
+  const double warningThreshold_;
+  const double throwingThreshold_;
+  const bool verbose_;
+  const bool dbFromEvent_;
+  const edm::EDGetTokenT<reco::BeamSpot> bsFromEventToken_;             // beamSpot from the event
+  edm::ESGetToken<BeamSpotObjects, BeamSpotObjectsRcd> bsFromDBToken_;  // beamSpot from the DB
+  edm::EDGetTokenT<reco::BeamSpot> dbBSfromEventToken_;                 // beamSpot from the DB (via event)
+};
+
+//
+// constructors and destructor
+//
+BeamSpotCompatibilityChecker::BeamSpotCompatibilityChecker(const edm::ParameterSet& iConfig)
+    : warningThreshold_(iConfig.getParameter<double>("warningThr")),
+      throwingThreshold_(iConfig.getParameter<double>("errorThr")),
+      verbose_(iConfig.getUntrackedParameter<bool>("verbose", false)),
+      dbFromEvent_(iConfig.getParameter<bool>("dbFromEvent")),
+      bsFromEventToken_(consumes<reco::BeamSpot>(iConfig.getParameter<edm::InputTag>("bsFromEvent"))) {
+  //now do what ever initialization is needed
+  if (warningThreshold_ > throwingThreshold_) {
+    throw cms::Exception("ConfigurationError")
+        << __PRETTY_FUNCTION__ << "\n Warning threshold (" << warningThreshold_
+        << ") cannot be smaller than the throwing threshold (" << throwingThreshold_ << ")" << std::endl;
+  }
+  if (dbFromEvent_) {
+    edm::LogWarning("BeamSpotCompatibilityChecker")
+        << "!!!! Warning !!!\nThe Database Beam Spot is going to be taken from the Event via BeamSpotProducer!";
+    dbBSfromEventToken_ = consumes<reco::BeamSpot>(iConfig.getParameter<edm::InputTag>("bsFromDB"));
+  } else {
+    bsFromDBToken_ = esConsumes<BeamSpotObjects, BeamSpotObjectsRcd>();
+  }
+}
+
+//
+// member functions
+//
+
+// ------------ method called for each event  ------------
+void BeamSpotCompatibilityChecker::analyze(edm::StreamID sid,
+                                           edm::Event const& iEvent,
+                                           edm::EventSetup const& iSetup) const {
+  using namespace edm;
+  reco::BeamSpot spotEvent, spotDB;
+
+  edm::Handle<reco::BeamSpot> beamSpotFromEventHandle;
+  iEvent.getByToken(bsFromEventToken_, beamSpotFromEventHandle);
+  spotEvent = *beamSpotFromEventHandle;
+
+  double evt_BSx0_ = spotEvent.x0();
```

Comment:
```suggestion
  double evt_BSx0 = spotEvent.x0();
```
to avoid confusion with data members

Reply:
fixed in the last push.